trading/trading/portfolio/lib/lots.mli

```diff
@@ -0,0 +1,90 @@
+(** Position lot tracking for accurate cost basis and P&L calculation.
+
+    This module provides lot-based position tracking with support for FIFO
+    (First In, First Out) and average cost accounting methods. *)
+
+open Trading_base.Types
+open Status
+
+type lot_id = string [@@deriving show, eq]
+(** Unique identifier for a position lot *)
+
+type position_lot = {
+  lot_id : lot_id;
+  quantity : quantity;  (** Can be positive (long) or negative (short) *)
+  cost_basis : float;
+      (** Total cost basis for entire lot including commission *)
+  acquisition_date : Time_ns_unix.t;  (** When acquired *)
+}
+[@@deriving show, eq]
+(** A single lot representing a portion of a position acquired at a specific
+    time and price. The cost_basis includes the total cost for all shares in
+    this lot, including commissions. *)
+
+(** Method for matching lots when closing positions *)
+type accounting_method =
+  | FIFO  (** First In, First Out - match oldest lots first (default) *)
+  | AverageCost  (** Weighted average cost - combines all lots *)
+[@@deriving show, eq]
+
+type position_with_lots = {
+  symbol : symbol;
+  lots : position_lot list;  (** All lots for this symbol *)
+  accounting_method : accounting_method;
+}
+[@@deriving show, eq]
+(** Position tracked by individual lots. Lots can have positive quantity (long)
+    or negative quantity (short). *)
+
+type lot_close_result = {
+  closed_lots : position_lot list;
+      (** Lots that were fully or partially closed *)
+  remaining_lots : position_lot list;  (** Lots remaining after close *)
+  realized_pnl : float;  (** Total realized P&L from closing *)
+}
+[@@deriving show, eq]
+(** Result of closing a position, showing which lots were affected and the P&L
+*)
+
+val create_lot :
```

Comment:
Is it necessary to have these `create_xyz` functions? Let's try go without them?